### DISCUSSION TITLE
Set local git repo config values on init

### DISCUSSION THREAD

**Discussion by old-pigeon** [twpayne/chezmoi#3683] [Q&A]
Hello,

I want to set config values specific to the chezmoi dotfile repo on init. I have two use cases. I have issues with both. Can you help?

Thank you!

## Windows CRLF checkout

On Windows, I want to checkout at least some files with enforced CRLF line endings for files which are sensitive to it. The following would need to run before chezmoi checks out the repo for the first time. So I thought about a `init` pre hook, but I think this is a problem because how would chezmoi even know that it should use that hook pre-init if the repo wasn't cloned in the first place? Feels like a chicken-egg-problem. What is the best approach to fix this? It would be too late to do it after initial checkout because the files are already wrongly written with only LN.

```
git config --local core.eol crlf
git config --local core.autocrlf true
```

## General user settings

I want to set the user settings for commit name and email for the repo once on any machine. I tried this to debug:

```
[hooks.init.post]
command = "git"
args = ["config", "--local", "user.email"]
```

The problem is that it is not executed inside the cloned chezmoi dotfile repo, it instead spits out this error:

```
fatal: --local can only be used inside a git repository
chezmoi: git: exit status 128
```

I tried inspecting all exported variables but none of them seem to point to the dir of the git repo reliable, so I can't use that either to pass it manually to the git command.

**Comment by twpayne** [comment]
> Windows CRLF checkout

For files that require Windows line endings, you can make them templates and use chezmoi's [`line-endings` template directives](https://www.chezmoi.io/reference/templates/directives/#line-endings). chezmoi will then fix the line endings for you.



> General user settings

Use a global git configuration file for this, i.e. create a file called `~/.config/git/config` containing

```
[user]
    email = my@email.com
```

and then manage this file with chezmoi.

**Comment by orzomaxx** [comment]
> Windows CRLF checkout

You can use `.gitattributes` file in the root of your git repository:
```gitattributes
* text=auto eol=lf

*.bat       text eol=crlf
*.bat.tmpl  text eol=crlf
desktop.ini text eol=crlf
```
For example, this makes all files to use LF, except `.bat` and `desktop.ini` files, which will use CRLF
